packages/flutter_tools/lib/src/asset.dart

```diff
@@ -530,16 +543,16 @@ class ManifestAssetBundle implements AssetBundle {
     // the uncompressed strings to not incur decompression/decoding while making
     // the comparison.
     if (!entries.containsKey(_kNoticeZippedFile) ||
-        (entries[_kNoticeZippedFile] as DevFSStringCompressingBytesContent?)
+        (entries[_kNoticeZippedFile]?.content as DevFSStringCompressingBytesContent?)
             ?.equals(combinedLicenses) != true) {
-      entries[_kNoticeZippedFile] = DevFSStringCompressingBytesContent(
+      entries[_kNoticeZippedFile] = AssetBundleEntry(
+        DevFSStringCompressingBytesContent(
```

Comment:
I think indentation in the following lines is now off? Adding a comma between close parens on line 555 might help.